pyscraper/committees/commons_api.py

```diff
@@ -0,0 +1,313 @@
+
+
+class Role(CamelModel):
+    start_date: datetime
+    end_date: Optional[datetime]
+    role: RoleDetails
+    ex_officio: bool
+    alternate: bool
+    co_opted: bool
+
+
+class MemberInfo(CamelModel):
+    member_from: Optional[str] = None
+    party: Optional[str] = None
+    party_colour: Optional[str] = None
+    mnis_id: int
+    is_chair: bool
+    list_as: str
+    display_as: str
+    full_title: str
+    address_as: Optional[str]
+    house: Optional[str] = None
+    is_current: bool
+
+
+class Member(CamelModel):
+    is_lay_member: bool
+    roles: List[Role]
+    id: int
+    person_id: Optional[int]
+    name: str
+    photo_url: str
+    member_info: Optional[MemberInfo] = None
+
+    def mnis_id(self):
+        if self.member_info and self.member_info.mnis_id:
+            return self.member_info.mnis_id
+
+
+class Contact(CamelModel):
+    email: Optional[str] = None
+    phone: Optional[str] = None
+    address: Optional[str] = None
+    contact_disclaimer: Optional[str] = None
+
+
+class Category(CamelModel):
+    id: int
+    name: str
+
+
+class CommitteeType(CamelModel):
+    id: int
+    name: str
+    committee_category: Category
+
+
+class NameHistory(CamelModel):
+    id: int
+    committee_id: int
+    name: str
+    start_date: datetime
+    end_date: Optional[datetime] = None
+
+
+class Department(CamelModel):
+    department_id: int
+    name: str
+
+
+class LeadHouse(CamelModel):
+    is_commons: bool
+    is_lords: bool
+
+
+class Committee(CamelModel):
+    id: int
+    name: str
+    house: str
+    purpose: Optional[str] = None
+    contact: Optional[Contact] = None
+    parent_committee: Optional[Committee] = None
+    sub_committees: Optional[List[Committee]] = None
+    lead_house: Optional[LeadHouse] = None
+    category: Optional[Category] = None
+    committee_types: List[CommitteeType]
+    show_on_website: bool
+    website_legacy_url: Optional[str] = None
+    website_legacy_redirect_enabled: bool
+    start_date: datetime
+    end_date: Optional[datetime] = None
+    date_commons_appointed: Optional[datetime] = None
+    date_lords_appointed: Optional[datetime] = None
+    scrutinising_departments: List[Department] = []
+    is_lead_committee: Optional[bool] = None
+    name_history: List[NameHistory]
+    members: List[Member] = Field(default_factory=list)
+
+    def parl_url(self):
+        return f"https://committees.parliament.uk/committee/{self.id}/"
+
+    def expand(self):
+        """
+        Get the full committee information from the API
+        """
+        url = f"https://committees-api.parliament.uk/api/Committees/{self.id}?includeBanners=false&showOnWebsiteOnly=false"
+        response = httpx.get(url, timeout=30)
+        response.raise_for_status()
+        data = response.json()
+        return Committee.model_validate(data).add_members()
+
+    def add_members(self):
+        """
+        Fetch and add members to the committee.
+        """
+
+        def one_request(skip=0):
```

Comment:
could we maybe call this `get_page_of_members` or something else more descriptive?